homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -287,7 +287,7 @@ def supported_features(self):
 
     async def async_pause(self):
         """Pause the cleaning task."""
-        if self.state == STATE_CLEANING:
+        if self.state == STATE_CLEANING or self.state == STATE_RETURNING:
```

Comment:
Do we need this check at all? Are there some limitations on the device that makes it required?

Reply:
in my opinion no, but I'm not the author of this check and cannot imagine any reason for it.

Reply:
I think it might be (or maybe it was like this before) that there was one command for start/pause so if you click pause multiple times then you will be switching between start and pause.
It is not like this right now.